Library/Formula/libphonenumber.rb

```diff
@@ -0,0 +1,37 @@
+class Libphonenumber < Formula
+  desc "C++ Phone Number library by Google"
+  homepage "https://github.com/googlei18n/libphonenumber"
+  url "https://github.com/googlei18n/libphonenumber/archive/libphonenumber-7.2.3.tar.gz"
+  sha256 "9f93615b23d8a12a42099d8113a8d6304278bee4d0ef09fd366188a206aacd46"
+
+  depends_on "cmake" => :build
+  depends_on :java => "1.7+"
+  depends_on "icu4c"
+  depends_on "protobuf"
+  depends_on "boost"
+  depends_on "re2"
+
+  patch :DATA
+
+  def install
+    system "cmake", "cpp", *std_cmake_args
+    system "make", "install"
+  end
+
```

Comment:
You could write a small test program and run it. See [`tinyxml2.rb`](https://github.com/Homebrew/homebrew/blob/e72bcfa571ed03a8490d0cdc587781b46c937e40/Library/Formula/tinyxml2.rb#L23-L33) for a good example.